.Package.test.swift

```diff
@@ -2,6 +2,10 @@ import PackageDescription
 
 let package = Package(
     name: "ReactiveSwift",
+    targets: [
+        Target(name: "ReactiveSwift", dependencies: ["OSLocking"]),
+        Target(name: "OSLocking")
+    ],
```

Comment:
I think it is not possible to move the rest of the C stuff to Swift though.

All uses of `_ras_*` function pointers in Swift are being resolved at link time. This is what the disassembly shows.

However, if written in Swift, the compiler would enforce availability, so have such pointers would trigger work at runtime. At best case, it would have to be a lazily initialised top-level (static) variable, which means extra overhead in obtaining the function pointer.

Until [SR-4079](https://bugs.swift.org/browse/SR-4079) is resolved, dropping down to C for performance is likely inevitable in this specific case.